### ISSUE TITLE
genPOI: Unable to identify banner base colour

### ISSUE THREAD

**Issue by jobe1986** [overviewer/Minecraft-Overviewer#1585]
As you know minecraft identifies base banner colours by their item id, however because genPOI translates ALL banner ids to minecraft:banner it is now no longer possible to identify the base colour of a banner. Pattern information is still present, but the base colour is not.

**Comment by CounterPillow** [comment]
>however because genPOI translates ALL banner ids to minecraft:banner

That's not genPOI doing it? It literally has 0 code for handling banners.

Looking in-game, the banner IDs are `minecraft:white_banner` and so on.

**Comment by jobe1986** [comment]
It's definately genPOI doing it.

Even in the NBT in the chunks banners are stored with an id of minecraft:<colour>_banner but in the poi structs its always an id of "minecraft:banner" regardless of colour.

**Comment by CounterPillow** [comment]
No, it's not Overviewer doing the translation. Minecraft writes two different NBT entries for banners:
1. an e.g. `minecraft:white_banner` entry in the block palette,
2. a `minecraft:banner` entry in TileEntities

genPOI *only* looks at TileEntities as searching through literally every block would be way too slow, it cannot look at blocks and their IDs. Mojang, in their infinite wisdom, decided that the banner colour should not be stored in TileEntities.

TL;DR: take this up with Mojang.
